src/ex1/script.js

```diff
@@ -0,0 +1,165 @@
+const inputTitle = document.getElementById("new-todo-title");
+const amountOfTodosInfo = document.getElementById("amount-info");
+
+const clearAllButton = document.getElementById("clear-all-button");
+const addTodoForm = document.getElementById("add-todo");
+const sortListButton = document.getElementById("sort-list-button");
+
+clearAllButton.addEventListener('click', onClearAllButtonClicked);
+addTodoForm.addEventListener('submit', onAddTodoFormSubmitted);
+sortListButton.addEventListener('click', onSortListButtonClicked);
+
+todosArray.forEach(todoText => {
+  addTodoItem(todoText, false);
+  displayFooterAndImage();
+});
+
+function addEventListenerForTodoTitle(listItem) {
+  const todoTitleDiv = listItem.getElementsByTagName("div")[0];
+  todoTitleDiv.addEventListener('click', ({target}) => {
+    onTodoTitleClicked(target);
+  });
+}
+
+function addEventListenerForDeleteButton(listItem) {
+  const deleteButton = listItem.getElementsByTagName("button")[0]
+  deleteButton.addEventListener('click', ({currentTarget}) => {
+    onDeleteButtonClicked(currentTarget);
+  });
+}
+
+function addTodoItem(todoText, animation) {
+  const listItem = createListElement(todoText, animation);
+  addEventListenerForTodoTitle(listItem);
+  addEventListenerForDeleteButton(listItem);
+  allTodosList.appendChild(listItem);
+  isListSorted = unsorted;
+}
+
+function createListElement(todoText, animation) {
+  const listItem = document.createElement("li");
+  if (animation) {
+    listItem.className = "add-item-animation";
+  }
+  setTimeout (() => { listItem.className = "existing-item";}, 700);
+  listItem.innerHTML = `<div class="todo-text">${todoText}</div>
+                        <button class="remove-todo-button"><i class="fa fa-trash"></i></button>`;
+  return listItem;
+}
+
+function displayFooterAndImage() {
+  displayButtonsAndAmount();
+  displayZeroImage();
+}
+
+function displayButtonsAndAmount() {
+  let tasks = "tasks";
+  if (todosArray.length === 1) {
+    tasks = "task";
+    displayElement('sort-list-button', false)
+    displayElement('clear-all-button', false)
+  } else {
+    displayElement('sort-list-button', true)
+    displayElement('clear-all-button', true)
+  }
+  amountOfTodosInfo.textContent = `${todosArray.length} pending ${tasks}`;
+}
+
+function displayZeroImage() {
+  if (todosArray.length === 0) {
+    displayElement('zero-todos-image', true)
+    displayElement('footer', false)
+  } else {
+    displayElement('zero-todos-image', false)
+    displayElement('footer', true)
+  }
+}
+
+function displayElement(elementId, toDisplay) {
+  displayStyle = toDisplay ? "" : "none";
+  document.getElementById(elementId).style.display = displayStyle;
+}
+
+function onTodoTitleClicked(clickedTodo) {
+  alert(clickedTodo.textContent);
+}
+
+function onDeleteButtonClicked(clickedButton) {
+  const index = Array.prototype.indexOf.call(allTodosList.getElementsByTagName("li"), clickedButton.parentElement);
+  todosArray.splice(index, 1);
```

Comment:
A cool technique you can use to decouple the animation code from the deletion code is to create a deleteTodoTask function and an animation wrapper function. ex:
function onDeleteButtonClicked(clickedButton) { deleteToDoTaskWithAnimation(); }

function deleteToDoTaskWithAnimation() {
  // animation code
  deleteTodoTask();
}

function deleteTodoTask() { // delete a todo code }

Reply:
same for the addition of tasks :)

Reply:
for deleting tasks I did in this way and for adding tasks I created a separated function with animation